src/carts/src/test/java/mushop/carts/test/TestCartService.java

```diff
@@ -1,171 +1,52 @@
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.client.RxHttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import mushop.carts.entitites.Cart;
+import mushop.carts.entitites.Item;
+import org.junit.jupiter.api.Test;
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.StringReader;
+import javax.inject.Inject;
 import java.math.BigDecimal;
-import java.net.HttpURLConnection;
-import java.net.InetAddress;
-import java.net.URL;
-import java.net.UnknownHostException;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
-import java.util.stream.Collectors;
+import java.util.List;
 
-import javax.json.Json;
-import javax.json.JsonArray;
-import javax.json.JsonObject;
-import javax.json.JsonReader;
-import javax.json.JsonValue;
-import javax.json.JsonWriter;
-import javax.json.bind.Jsonb;
-import javax.json.bind.JsonbBuilder;
-
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-
-import io.helidon.webserver.WebServer;
-import mushop.carts.Cart;
-import mushop.carts.Item;
-import mushop.carts.Main;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 
+@MicronautTest
 public class TestCartService {
-    
-    WebServer server;
-    
-    @BeforeEach
-    public void setUp() throws Exception {
-        server = Main.createWebServer();
-        server.start().toCompletableFuture().join();
-    }
 
-    @AfterEach
-    public void tearDown() throws Exception {
-        server.shutdown().toCompletableFuture().join();
-        server = null;
-    }
-    
-    @Test
-    public void testMetricsJson() throws Exception {
-        JsonObject result = get(baseUrl() + "/metrics", "Accept: application/json").asJsonObject();
-
-        assertTrue(result.get("vendor").asJsonObject()
-                         .getInt("requests.count") > 0);
-        
-    }
+    @Inject
+    @Client("/")
+    RxHttpClient httpClient;
 
     @Test
-    public void testMetricsText() throws Exception {
-        String result = getPlainText(baseUrl() + "/metrics", "Accept: text/plain");
-        Boolean result_b = Arrays.stream(result.split(System.lineSeparator()))
-                .filter(line -> !line.startsWith("#"))
-                //.map(line -> {System.out.println(line); return line;})
-                .map(line -> line.matches("([a-zA-Z_:]*)(.*)[ ]([0-9.]*)"))
-                .allMatch(bool -> bool==true);
-        System.out.println(result_b);;
-
-        assertTrue(result_b);
-
-    }
-
-    @Test
-    public void testHealthCheck() throws Exception {
-        JsonValue result = get(baseUrl() + "/health");
-        assertEquals("UP", result.asJsonObject().getString("status"));
-    }
```

Comment:
Why were the health and metrics tests deleted?

Reply:
I didn't want to test smth that is provided by micronaut. But I can move them back